dotCMS/src/main/java/com/dotmarketing/quartz/job/CascadePermissionsJob.java

```diff
@@ -115,7 +115,7 @@ public void execute(JobExecutionContext jobContext) throws JobExecutionException
 			permissionable = (Permissionable) retrievePermissionable(permissionableId);
 			Role role = (Role) roleAPI.loadRoleById(roleId);
 			permissionAPI.cascadePermissionUnder(permissionable, role);
-			permissionAPI.removePermissionableFromCache(permissionable.getPermissionId());
+			permissionAPI.clearCache();

```

Comment:
We should never have to do a full permissions cache flush.  Think about customers who have 1000's of sites with 1000's pages.  This is too broad.

Reply:
@wezell what about this one? https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/business/PermissionBitFactoryImpl.java#L1689 . This is the logic that runs when permissions are updated from the site properties instead of "Roles & Tools"

Reply:
@wezell do you want to take a look at my new changes? I didn't include the unification of the reset/cascade logic because it might affect important logic and implies a heavy refactoring. A first attempt was done here: https://github.com/dotCMS/core/pull/10434/files